src/authentication/authentication.service.ts

```diff
@@ -111,4 +116,30 @@ export class AuthenticationService {
       },
     };
   }
+
+  async forgotPassword(
+    forgotPasswordDto: ForgotPasswordDto,
+  ): Promise<IResponse<User>> {
+    const user = await this.usersService.findByEmail(forgotPasswordDto.email);
+    if (!user) {
+      throw new NotFoundException('User with email does not exist');
```

Comment:
Since you're thinking about the users maybe we can add the actual email for clarify lol
```suggestion
      throw new NotFoundException(`User with email ${email} does not exist`);
```